# Unit 4 Project

In this project, you will summarize and present your analysis from Projects 1-3.

### Intro: Write a problem Statement/ Specific Aim for this project

How does an undergraduate's college rank a affect their likelihood of graduate school admission?

### Dataset:  Write up a description of your data and any cleaning that was completed

The admissions dataset we are exploring includes 400 observations with four features: admit, gre, gpa, and prestige (rank of student's undergraduate alma mater).

Variable | Description | Type of Variable
---| ---| ---
admit | 0 = not admitted 1 = admitted | categorical
gre | GRE score | continuous
gpa | grade point average | continuous
prestige | 1 = highest 4 = lowest | categorical

To clean the data, we dropped observations with missing data points, and then created dummy variables for prestige.

### Demo: Provide a table that explains the data by admission status

In [37]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm
import pylab as pl
import numpy as np
%matplotlib inline

In [38]:
df_raw = pd.read_csv("admissions.csv")
df = df_raw.dropna() 

In [39]:
#gre and gpa mean
df.groupby("admit").mean()

gre       gpa  prestige
admit                                
0      573.579336  3.347159  2.645756
1      618.571429  3.489206  2.150794

In [40]:
#gre and gpa standard deviation
df.groupby("admit").std()

gre       gpa  prestige
admit                                
0      116.052798  0.376355  0.918922
1      109.257233  0.371655  0.921455

In [41]:
# frequency table for prestige and whether or not someone was admitted
prestige_admit = pd.crosstab(index=df["admit"], 
                           columns=df["prestige"])

prestige_admit.index= ["not admitted","admitted"]

prestige_admit

prestige      1.0  2.0  3.0  4.0
not admitted   28   95   93   55
admitted       33   53   28   12

Mean (STD) or counts by admission status for each variable 

| Not Admitted | Admitted
---| ---|---
GPA | 3.35(0.37)  | 3.49(0.37)
GRE | 574(116) | 619(109)
Prestige 1 | 46% | 54%
Prestige 2 | 64% | 36%
Prestige 3 | 77% | 23%
Prestige 4 | 82% | 18%

### Methods: Write up the methods used in your analysis

First we fit the data to a logistic regression model with Statsmodel. We dropped prestige with rank 1 to avoid colinearity in our model.

To evaluate the model, we recreated the dataset with every logical combination of input values, to see how the predicted probability of admission increases/decreases across different variables.

### Results: Write up your results

Students who attended a #2 ranked undergaduate college had 0.50 the odds of being admitted to graduate school, compared to students who attended a #1 ranked undergaduate college.

Students who attended a #3 ranked undergaduate college had 0.26 the odds of being admitted to graduate school, compared to students who attended a #1 ranked undergaduate college.

Students who attended a #4 ranked undergaduate college had 0.21 the odds of being admitted to graduate school, compared to students who attended a #1 ranked undergaduate college.

Odds of admission increase by 118% for every 1.0 increase in GPA.

Odds of admission increae by 22% for ever 100-unit increase in GRE score.

### Visuals: Provide a table or visualization of these results

<img src='prestige and admit.png' height= 35% width= 35%>

<img src='gpa and prestige.png' height= 45% width= 45%>

<img src='gre and prestige.png' height= 45% width= 45%>

### Discussion: Write up your discussion and future steps

There's a positive trend for the effect of GPA, GRE scores, and prestige on an applicant's probability of admission into grad school. For exactly the same GPA or GRE scores, applicants coming from more prestigious schools will have a higher probability of admission.  There's a larger drop in probability of admission between a prestige rank of 1 and 2 than between 2 and 3 or 3 and 4.

In the future, I would improve upon this model by using the sk learn package to standardize features. GRE had a very large standard deviation compared to GPA. I would also evaluate the model's mean squared error. 